extensions/amp-viewer-integration/0.1/test/test-touch-handler.js

```diff
@@ -114,21 +114,59 @@ describes.fakeWin('TouchHandler', {}, (env) => {
       expect(messages[0].data.data.type).to.equal('touchstart');
     });
 
-    it('should not forward events marked with `shouldViewerCancelPropagation`', () => {
-      const event = fakeTouchEvent('touchmove');
-      event.stopImmediatePropagation = env.sandbox.spy();
+    describe('event propagation', () => {
+      let event, event2;
+
+      beforeEach(() => {
+        event = fakeTouchEvent('touchmove');
+        event2 = fakeTouchEvent('touchmove');
+        event.stopImmediatePropagation = env.sandbox.spy();
+        event2.stopImmediatePropagation = env.sandbox.spy();
+      });
 
-      touchHandler.handleEvent_(event);
-      expect(messages).to.have.length(1);
-      expect(messages[0].data.data.type).to.equal('touchmove');
-      expect(event.stopImmediatePropagation).to.not.be.called;
+      it('should forward events if event does not contain `shouldViewerCancelPropagation`', () => {
+        touchHandler.handleEvent_(event);
 
-      event.shouldViewerCancelPropagation = true;
-      messages = [];
-      touchHandler.handleEvent_(event);
+        expect(messages).to.have.length(1);
+        expect(messages[0].data.data.type).to.equal('touchmove');
```

Comment:
FMI: Is it worth adding a code comment for why there is two `data` accesses? May not be obvious to reader.

Reply:
It's a nuance of the test since we mock the windows post message.